v2/client.go

```diff
@@ -363,6 +382,20 @@ func NewPointFrom(pt models.Point) *Point {
 	return &Point{pt: pt}
 }
 
+func gzipReader(r io.Reader) io.Reader {
+	pr, pw := io.Pipe()
+	go func() {
+		gzw := gzip.NewWriter(pw)
+		if _, err := io.Copy(gzw, r); err != nil {
+			pw.CloseWithError(err)
```

Comment:
@ayang64 does `gzw` need to be closed here as well?

Reply:
> @ayang64 does `gzw` need to be closed here as well?

it should be. I updated it with a defer call